packages/stream/components/utils.ts

```diff
@@ -0,0 +1,32 @@
+type PendingThenableState = { status: "pending" };
+type FulfilledThenableState<T> = { status: "fulfilled"; value: T };
+type RejectedThenableState = { status: "rejected"; reason: unknown };
+
+type ThenableState<T> =
```

Comment:
nit: wondering if Promise would be a better name than `Thenable` for these variables?
- PendingPromiseState
- FulfilledPromiseState
...

I understand we want to differentiate from standard promises. Though I'm not sure how much the name "thenable" is gaining us here. Took me a second to equate the two.

Reply:
TBH i basically copied React's naming and conventions around this wholesale :D
https://github.com/facebook/react/blob/54f2314e9cfe82baf1c040a55ed4dfff6488f84e/packages/shared/ReactTypes.js/#L126-L148
could change it to a promise, but i guess that it's some sort of precedent, so i thought i'd just stick with that

Reply:
Ah, good to know there's presidence. To be honest, I'm still not sold on the naming, but I don't feel strongly enough about it to block on changes. We can keep this